src/snapred/backend/dao/DetectorPeak.py

```diff
@@ -0,0 +1,9 @@
+from pydantic import BaseModel
+
+
+class DetectorPeak(BaseModel):
+    """Class to hold diffraction peak position and limits in d-spacing"""
+
+    position: float
+    limLeft: float
+    limRight: float
```

Comment:
Id replace these two with a `Limit[float]`

Reply:
Yes, I thought about it.  Actually there is more to the story: we want `limLeft <= position <= limRight`.  So maybe we need an extended version of the `Limit` class?

Reply:
well, these variables would become `positionLimit`.  What's the additional usecase?  You can add a validator to this object to ensure that position satisfies this requirement. [See this example.](https://github.com/neutrons/SNAPRed/blob/db74e5964e9b2527d662d7277cdd64a56c474b21/src/snapred/backend/dao/FitMultiplePeaksIngredients.py#L2)

Reply:
Which I suppose you could then extract out to some sorted of `BoundedValue[float]` generic object

Reply:
Done.